lib/kitchen/provisioner/chef_base.rb

```diff
@@ -167,22 +173,34 @@ def install_options
       #   kitchen root
       # @api private
       def policyfile
-        policyfile_basename = config[:policyfile_path] || "Policyfile.rb"
-        File.join(config[:kitchen_root], policyfile_basename)
+        if [nil, "policyfile", :policyfile].include?(config[:preferred_resolver])
```

Comment:
Instead of checking the preferred resolver each time in `policyfile`, `berksfile`, and `cheffile`, maybe we could resolve the preferred provider earlier and leave these methods alone?